src/org/labkey/test/io/Grep.java

```diff
@@ -51,6 +54,18 @@
 
     private static int grep(File file, Pattern pattern) throws IOException
     {
+        final List<Pair<String, Integer>> matches = findMatches(file, pattern, true);
+
+        return matches.stream().findFirst().map(Pair::getRight).orElse(-1);
+    }
+
+    /**
+     * Find all matches to the given pattern. Also includes line numbers for the given pattern.
+     */
+    public static List<Pair<String, Integer>> findMatches(File file, Pattern pattern, boolean firstMatch) throws IOException
+    {
+        List<Pair<String, Integer>> matches = new ArrayList<>();
+
         try (BufferedReader is = Readers.getReader(file))
```

Comment:
Pre-existing code, but should this be `br` intead of `is`? Also, there is `BufferedReader.lines()`, which returns a `Stream<String>`. Maybe not as handy in this specific case because of the short-circuit and line numbers. And the fact that what you have works fine. 😄